### PR TITLE
Fix bugs of Slice/ReduceSum/TopK nodes and Graph's member function in corner cases

### PR DESCRIPTION
**Pull request by Treemann**
Hi, I found some bugs and fixed them when processing a complicated case. The description of this pull requrest is: 

> Fix bugs of Slice/ReduceSum/TopK nodes in corner cases and optimize the member function (\_\_find_shape_tensor\_\_) of the Graph class

You could learn the details from the four commits history.

### CODE REVIEW

**Comment by ThanatosShinji** [comment]
Hi, thanks for your contribution! Can you provide onnx models or pytorch(tf) scripts to test your fix?

**Comment by Treemann** [comment]
Hi, subjected to the license, sorry that I could not send the model to you. 
But I could provide snapshots of the nodes related for your verification. I think the bug-fix is clear and easy to understand with these snapshots, you may test the code with your self-defined model if possible, or you could communicate with me if there is any question.
slice
reduceSum
topk

**Comment by ThanatosShinji** [comment]
Thanks again! I will use your fork repo to test all archived models. Then I will find a way to test these corner cases.

BTW, if it is convenient for you, you can use the ['extract subgraph'](https://github.com/ThanatosShinji/onnx-tool/blob/main/data/Subgraph.md) function to get the subgraph with these corner cases from your private models.  I think there is no secret in these subgraphs. Of course, it's all up to you.

**Comment by Treemann** [comment]
Sorry for the inconvenience of uploading any subgraph to the internet~
I could explain the modification of each commit.
(1) Slice: like the snapshot of Slice node I provided, starts=[0,0,0,6], ends=[2147483647, 2147483647, 2147483647, 7], the order of both starts & ends are correspondent with the order of axes=[0,2,3,1]. So for axis=1, start=6, end=7; but the original code visit axes with the order of [0,1,2,3] instead of [0,2,3,1], then causes error.
(2) ReduceSum: like the snapshot of ReduceSum node I provided, if the node has two inputs, the second one is the list of axes. I notice that in node.py you have consider this situation, but in node_profiles.py you only consider the case of one input.
(3) TopK: like the snapshot of TopK node I provided, this node doesn't have the axis attribute, then self.axis=None. Since the input tensor has only one dimension, and the input shape of this node is (4000,), so TopK is conduced along axis=0 although    there is no axis attribute here.
(4) On the 4th commit [00d9a48](https://github.com/ThanatosShinji/onnx-tool/pull/15/commits/00d9a48ac23919e0bdc5b68894130ad766f35260), refer to the commit message and the part of my modification should be recovered to your original version according to our discussion on the issue.

File: onnx_tool/graph.py [hunk 1]
```diff
@@ -199,11 +199,12 @@ def __find_shape_tensors__(self):
                     for nname in searchnodes:
                         node = self.nodemap[nname]
                         node.shape_calc = True
-                        if node.op_type == 'Shape':
-                            continue
+                        # if node.op_type == 'Shape':
```
Comment (ThanatosShinji) [inline onnx_tool/graph.py:202]: these codes need to be reverted.

**Review by ThanatosShinji — CHANGES_REQUESTED** [review]

File: onnx_tool/graph.py (hunk 1 above)
Comment (Treemann) [reply onnx_tool/graph.py]: Done, please check the latest commit!

**Review by ThanatosShinji — APPROVED** [review]
All model tests passed.

**Comment by ThanatosShinji** [comment]
Good work!
